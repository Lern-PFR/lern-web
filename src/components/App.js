function App() {
  return (
    <div className="App">
      <header className="App-header">
        <p>
          Edit <code>src/components/App.js</code> and save to reload.
        </p>
      </header>
    </div>
  );
}

export default App;
