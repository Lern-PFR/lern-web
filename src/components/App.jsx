import { Home } from 'react-feather';
import { MainLayout, StyledDiv } from 'components/shared/layout';
import {
	Canon,
	Trafalgar,
	Paragon,
	GreatPrimer,
	DoublePica,
	BodyCopy,
	Pica,
	LongPrimer,
	Brevier,
	Minion,
} from 'components/shared/typography';
import {
	PrimaryButton,
	StandardButton,
	SubtleButton,
	DropdownButton,
	DangerButton,
	IconButton,
	FloatingActionButton,
} from 'components/shared/buttons';
import {
	elevationFlat,
	elevationFirst,
	elevationSecond,
	elevationThird,
	elevationFourth,
	elevationFifth,
} from 'theme/elevations';
import {
	LabeledCheckbox,
	FieldsetComponent,
	LabeledRadioButton,
	LabelComponent,
	LabeledInput,
} from 'components/shared/form';

/**
 * @name App
 * @description The main component of the application.
 *
 * @author TimothÃ©e Simon-Franza
 */
const App = () => (
	<MainLayout>
		<StyledDiv position="relative">
			<Canon>Canon text element.</Canon>
			<Trafalgar>Trafalgar text element.</Trafalgar>
			<Paragon>Paragon text element.</Paragon>
			<GreatPrimer>GreatPrimer text element.</GreatPrimer>
			<DoublePica>DoublePica text element.</DoublePica>
			<BodyCopy>BodyCopy text element.</BodyCopy>
			<Pica>Pica text element.</Pica>
			<LongPrimer>LongPrimer text element.</LongPrimer>
			<Brevier>Brevier text element.</Brevier>
			<Minion>Minion text element.</Minion>
		</StyledDiv>
		<StyledDiv position="relative">
			<PrimaryButton>Primary button</PrimaryButton>
			<StandardButton>Standard button</StandardButton>
			<SubtleButton>Subtle button</SubtleButton>
			<DropdownButton>Dropdown button</DropdownButton>
			<DangerButton>Danger button</DangerButton>
			<IconButton><Home /></IconButton>
			<FloatingActionButton isIconOnly><Home /></FloatingActionButton>
		</StyledDiv>
		<StyledDiv position="relative" marginTop="50px" display="flex" width="100%">
			<StyledDiv {...elevationFlat} width="260px" height="260px" padding="10px">Flat component</StyledDiv>
			<StyledDiv {...elevationFirst} width="260px" height="260px" padding="10px">5% elevation component</StyledDiv>
			<StyledDiv {...elevationSecond} width="260px" height="260px" padding="10px">10% elevation component</StyledDiv>
			<StyledDiv {...elevationThird} width="260px" height="260px" padding="10px">20% elevation component</StyledDiv>
			<StyledDiv {...elevationFourth} width="260px" height="260px" padding="10px">30% elevation component</StyledDiv>
			<StyledDiv {...elevationFifth} width="260px" height="260px" padding="10px">50% elevation component</StyledDiv>
		</StyledDiv>
		<StyledDiv position="relative" display="flex" flexDirection="column">
			<LabelComponent>Default label component</LabelComponent>
			<LabelComponent textStyle="bodycopy">BodyCopy label component</LabelComponent>
		</StyledDiv>
		<StyledDiv position="relative">
			<FieldsetComponent title="Basic checkbox components">
				<LabeledCheckbox id="lcb5">Labeled checkbox</LabeledCheckbox>
				<LabeledCheckbox labelTextStyle="bodycopy" checked id="lcb6">Default checked labeled checkbox</LabeledCheckbox>
				<LabeledCheckbox disabled id="lcb7">Disabled labeled checkbox</LabeledCheckbox>
				<LabeledCheckbox disabled checked id="lcb8">Disabled default checked labeled checkbox</LabeledCheckbox>
			</FieldsetComponent>
		</StyledDiv>
		<StyledDiv position="relative">
			<FieldsetComponent title="Basic radio button components">
				<LabeledRadioButton name="rg2" id="lrb5">Labeled radio button</LabeledRadioButton>
				<LabeledRadioButton labelTextStyle="bodycopy" name="rg2" checked id="lrb6">Default checked labeled radio button</LabeledRadioButton>
				<LabeledRadioButton name="rg2" disabled id="lrb7">Disabled labeled radio button</LabeledRadioButton>
			</FieldsetComponent>
		</StyledDiv>
		<StyledDiv position="relative" width="500px">
			<FieldsetComponent title="Text input components">
				<LabeledInput id="basicTextInput">Labeled text input</LabeledInput>
				<LabeledInput id="placeholderTextInput" placeholder="Enter details here.">Labeled text input with custom placeholder</LabeledInput>
				<LabeledInput id="hintTextInput" placeholder="Enter details here." hintText="This is a hint for the user.">Labeled text input with hint</LabeledInput>
				<LabeledInput
					id="disabledTextInput"
					placeholder="Enter details here."
					hintText="This is a hint for the user."
					disabled
				>
					Disabled labeled text input
				</LabeledInput>
				<LabeledInput
					id="errorTextInput"
					placeholder="Enter details here."
					hintText="This is a hint for the user."
					errorText="There is something wrong with the value in this input"
					error
				>
					Labeled text input with a validation error
				</LabeledInput>
				<LabeledInput id="numberInput" placeholder="Number Input" type="number">Labeled number input</LabeledInput>
				<LabeledInput id="passwordInput" placeholder="Password Input" type="password">Labeled password input</LabeledInput>
			</FieldsetComponent>
		</StyledDiv>
	</MainLayout>
);

export default App;
