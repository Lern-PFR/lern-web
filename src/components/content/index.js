import ModuleCreationForm from './ModuleCreationForm';
import ModuleEditionForm from './ModuleEditionForm';
import ModuleConceptList from './ModuleConceptList';
import NavigationSidebar from './NavigationSidebar';
import SubjectCreationForm from './SubjectCreationForm';
import SubjectEditionForm from './SubjectEditionForm';
import SubjectModuleList from './SubjectModuleList';

export {
	ModuleCreationForm,
	ModuleEditionForm,
	ModuleConceptList,
	NavigationSidebar,
	SubjectCreationForm,
	SubjectEditionForm,
	SubjectModuleList,
};
