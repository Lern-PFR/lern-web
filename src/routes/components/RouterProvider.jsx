import PropTypes from 'prop-types';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';

/**
 * @name history
 * @description The global history object, handling the navigation for the whole application
 *
 * @author Yann Hodiesne
 */
export const history = createBrowserHistory();

/**
 * @name RouterProvider
 * @description A component providing a Router from react-router, with the ability to export its history and block navigation
 *
 * @author Yann Hodiesne
 *
 * @param {*} children : The children to render inside the Router
 */
const RouterProvider = ({ children }) => (
	<Router history={history}>
		{children}
	</Router>
);

RouterProvider.propTypes = {
	children: PropTypes.node.isRequired,
};

export default RouterProvider;
