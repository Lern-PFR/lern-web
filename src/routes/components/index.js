import RenderRoutes from './RenderRoutes';

export {
	// TODO: disable this exception once there is more to export
	// eslint-disable-next-line import/prefer-default-export
	RenderRoutes,
};
