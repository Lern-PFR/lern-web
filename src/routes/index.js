import keys from './keys';

export {
	keys as default,
};
