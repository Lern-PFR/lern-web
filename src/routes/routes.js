/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/display-name */
import { PostSignupPage, SignInPage, SignupPage } from 'pages/auth';
import { ConceptCreationPage, ConceptDetailsPage, ConceptEditionPage } from 'pages/concepts';
import { SubjectCreationPage, SubjectDetailsPage, SubjectEditionPage, SubjectListPage, SubjectStructurePage } from 'pages/subjects';
import { ModuleCreationPage, ModuleDetailsPage, ModuleEditionPage } from 'pages/modules';
import { Homepage } from 'pages/home';
import { LessonCreationPage } from 'pages/lessons';
import keys from './keys';

/**
 * Routes to render
 * For authentication purposes, each route can have `allowAnonymous` and `allowAuthenticated` values
 * Default values are :
 *  - `allowAnonymous`		: false
 *  - `allowAuthenticated`	: true
 */
export default [
	{
		path: keys.app.default,
		exact: true,
		allowAnonymous: true,
		component: Homepage,
	},
	{
		path: keys.about.default,
		exact: true,
		allowAnonymous: true,
		allowAuthenticated: true,
		component: () => <p>About</p>,
	},
	{
		path: keys.contactUs.default,
		exact: true,
		allowAnonymous: true,
		allowAuthenticated: true,
		component: () => <p>Contact us</p>,
	},
	// Home
	{
		path: keys.home.default,
		exact: true,
		allowAnonymous: true,
		allowAuthenticated: true,
		component: Homepage,
	},
	// Auth
	{
		path: keys.auth.default,
		routes: [
			{
				path: keys.auth.login,
				allowAnonymous: true,
				allowAuthenticated: false,
				exact: true,
				component: SignInPage,
			},
			{
				path: keys.auth.logout,
				exact: true,
				component: () => <p>Logout</p>,
			},
			{
				path: keys.auth.forgottenPassword,
				allowAnonymous: true,
				allowAuthenticated: false,
				exact: true,
				component: () => <p>Forgot password</p>,
			},
			{
				path: keys.auth.signup,
				allowAnonymous: true,
				allowAuthenticated: false,
				exact: true,
				component: SignupPage,
			},
			{
				path: keys.auth.postSignup,
				allowAnonymous: true,
				allowAuthenticated: false,
				exact: true,
				component: PostSignupPage,
			},
		],
	},
	// Subjects
	{
		path: keys.subjects.default,
		routes: [
			{
				path: keys.subjects.list,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: SubjectListPage,
			},
			{
				path: keys.subjects.subjectStructure,
				allowAnonymous: true,
				allowAuthenticated: true,
				exact: true,
				component: SubjectStructurePage,
			},
			{
				path: keys.subjects.subjectCreation,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: SubjectCreationPage,
			},
			{
				path: keys.subjects.subjectEdition,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: SubjectEditionPage,
			},
			{
				path: keys.subjects.moduleCreation,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ModuleCreationPage,
			},
			{
				path: keys.subjects.subjectDetails,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: SubjectDetailsPage,
			},
		],
	},
	// Modules
	{
		path: keys.modules.default,
		routes: [
			{
				path: keys.modules.moduleEdition,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ModuleEditionPage,
			},
			{
				path: keys.modules.conceptCreation,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ConceptCreationPage,
			},
			{
				path: keys.modules.moduleDetails,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ModuleDetailsPage,
			},
		],
	},
	// Concepts
	{
		path: keys.concepts.default,
		routes: [
			{
				path: keys.concepts.conceptEdition,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ConceptEditionPage,
			},
			{
				path: keys.concepts.conceptDetails,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ConceptDetailsPage,
			},
			{
				path: keys.concepts.lessonCreation,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: LessonCreationPage,
			},
			{
				path: keys.concepts.lessonDetails,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ConceptDetailsPage,
			},
		],
	},
	// Lessons
	{
		path: keys.lessons.default,
		routes: [
			{
				path: keys.lessons.lessonEdition,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: () => <p>Lesson edition</p>,
			},
		],
	},
];
