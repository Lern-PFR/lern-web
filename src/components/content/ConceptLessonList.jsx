import { useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';
import { PlusCircle } from 'react-feather';
import { generatePath } from 'react-router';
import styled from 'styled-components';

import routes from 'routes';
import { StyledDiv, StyledList, StyledListItem } from 'components/shared/styledElements';
import { DoublePica } from 'components/shared/typography';
import { history } from 'routes/components/RouterProvider';

import { addInnerContentCard, innerContentList, innerContentListItem } from 'theme/contentEditionCommon/innerContentListTheme';
import ConceptLessonListItem from './ConceptLessonListItem';

const ListCard = styled(StyledListItem)(
	{
		...innerContentListItem,
	}
);

/**
 * @name ConceptLessonList
 * @description A component used in the concept edition page to display its lessons and a "create new lesson" link.
 *
 * @author TimothÃ©e Simon-Franza
 *
 * @param {Array}	lessonList	The lesson list of the concept currently being edited.
 * @param {string}	conceptId	The id of the parent concept.
 */
const ConceptLessonList = ({ lessonList, conceptId }) => {
	const { t } = useTranslation();

	const onAddLessonCardClick = useCallback(() => {
		history.push(generatePath(routes.concepts.lessonCreation, { conceptId }));
	}, [conceptId]);

	return (
		<StyledDiv marginTop="64px">
			<DoublePica as="h3">{t('concepts.edition.lessons_list.title')}</DoublePica>
			<StyledList {...innerContentList}>
				{lessonList.map((lesson) => (
					<ConceptLessonListItem key={lesson.id} {...lesson} />
				))}
				<ListCard {...addInnerContentCard} onClick={onAddLessonCardClick}>
					<PlusCircle />
					{t('concepts.edition.lessons_list.links.add')}
				</ListCard>
			</StyledList>
		</StyledDiv>
	);
};

ConceptLessonList.propTypes = {
	lessonList: PropTypes.arrayOf(
		PropTypes.shape({
			description: PropTypes.string.isRequired,
			id: PropTypes.string.isRequired,
			title: PropTypes.string.isRequired,
		}),
	).isRequired,
	conceptId: PropTypes.string.isRequired,
};

export default ConceptLessonList;
