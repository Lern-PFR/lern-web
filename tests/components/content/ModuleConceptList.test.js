import { shallow } from 'enzyme';
import { act, fireEvent, render, screen } from '@testing-library/react';

import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { ModuleConceptList } from 'components/content';
import routes from 'routes';
import { history } from 'routes/components/RouterProvider';

describe('Module edition page - concept list', () => {
	const mockStore = configureMockStore([thunk]);
	const store = mockStore({});

	afterEach(() => {
		jest.restoreAllMocks();
		jest.resetAllMocks();
	});

	describe('snapshots', () => {
		it('should match previous snapshot with an empty moduleList prop', () => {
			const sut = (
				<Provider store={store}>
					<ModuleConceptList conceptList={[]} moduleId="1" subjectId="1" />
				</Provider>
			);
			const wrapper = shallow(sut);

			expect(wrapper).toMatchSnapshot();
		});

		it('should match previous snapshot with a non-empty moduleList prop', () => {
			const mockedModuleList = [
				{ id: 'dummy_module_id_1', title: 'dummy_module_title_1', description: 'dummy_module_desc_1' },
				{ id: 'dummy_module_id_2', title: 'dummy_module_title_2', description: 'dummy_module_desc_2' },
				{ id: 'dummy_module_id_3', title: 'dummy_module_title_3', description: 'dummy_module_desc_3' },
			];

			const sut = (
				<Provider store={store}>
					<ModuleConceptList conceptList={mockedModuleList} moduleId="1" subjectId="1" />
				</Provider>
			);
			const wrapper = shallow(sut);

			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('onClick events', () => {
		it('should call history.push with the module creation route upon click on the module creation card.', () => {
			const historyPushSpy = jest.spyOn(history, 'push');
			render(
				<Provider store={store}>
					<ModuleConceptList conceptList={[]} moduleId="1" subjectId="1" />
				</Provider>
			);

			act(() => {
				fireEvent.click(screen.getByRole('listitem'));
			});

			expect(historyPushSpy).toHaveBeenCalledTimes(1);
			expect(historyPushSpy).toBeCalledWith(routes.modules.conceptCreation.replace(':moduleId', '1'));
		});
	});
});
