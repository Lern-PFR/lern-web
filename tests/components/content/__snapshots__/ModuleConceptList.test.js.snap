// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module edition page - concept list snapshots should match previous snapshot with a non-empty moduleList prop 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ModuleConceptList
    conceptList={
      Array [
        Object {
          "description": "dummy_module_desc_1",
          "id": "dummy_module_id_1",
          "title": "dummy_module_title_1",
        },
        Object {
          "description": "dummy_module_desc_2",
          "id": "dummy_module_id_2",
          "title": "dummy_module_title_2",
        },
        Object {
          "description": "dummy_module_desc_3",
          "id": "dummy_module_id_3",
          "title": "dummy_module_title_3",
        },
      ]
    }
    moduleId="1"
    subjectId="1"
  />
</ContextProvider>
`;

exports[`Module edition page - concept list snapshots should match previous snapshot with an empty moduleList prop 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ModuleConceptList
    conceptList={Array []}
    moduleId="1"
    subjectId="1"
  />
</ContextProvider>
`;
